src/dex/maker-psm/maker-psm.ts

```diff
@@ -0,0 +1,514 @@
+  protected eventPools: { [gemAddress: string]: MakerPsmEventPool };
+
+  // warning: There is limit on swap
+  readonly hasConstantPriceLargeAmounts = true;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(MakerPsmConfig);
+
+  logger: Logger;
+
+  constructor(
+    protected network: Network,
+    protected dexKey: string,
+    protected dexHelper: IDexHelper,
+    protected adapters = Adapters[network],
+    protected dai: Token = MakerPsmConfig[dexKey][network].dai,
+    protected vatAddress: Address = MakerPsmConfig[dexKey][network].vatAddress,
+    protected poolConfigs: PoolConfig[] = MakerPsmConfig[dexKey][network].pools,
+  ) {
+    super(dexHelper.augustusAddress, dexHelper.provider);
+    this.logger = dexHelper.getLogger(dexKey);
+    this.eventPools = {};
+    poolConfigs.forEach(
+      p =>
+        (this.eventPools[p.gem.address.toLowerCase()] = new MakerPsmEventPool(
+          dexKey,
+          network,
+          dexHelper,
+          this.logger,
+          p,
+          this.vatAddress,
+        )),
+    );
+  }
+
+  async initializePricing(blockNumber: number) {
+    const poolStates = await getOnChainState(
+      this.dexHelper.multiContract,
+      this.poolConfigs,
+      this.vatAddress,
+      blockNumber,
+    );
+    this.poolConfigs.forEach((p, i) => {
+      const eventPool = this.eventPools[p.gem.address.toLowerCase()];
+      eventPool.setState(poolStates[i], blockNumber);
+      this.dexHelper.blockManager.subscribeToLogs(
+        eventPool,
+        eventPool.addressesSubscribed,
+        blockNumber,
+      );
+    });
+  }
+
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side];
+  }
+
+  getEventPool(srcToken: Token, destToken: Token): MakerPsmEventPool | null {
+    const srcAddress = srcToken.address.toLowerCase();
+    const destAddress = destToken.address.toLowerCase();
+    return (
+      (srcAddress === this.dai.address && this.eventPools[destAddress]) ||
+      (destAddress === this.dai.address && this.eventPools[srcAddress]) ||
+      null
+    );
+  }
+
+  // Returns list of pool identifiers that can be used
+  // for a given swap. poolIdentifers must be unique
+  // across DEXes.
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    const eventPool = this.getEventPool(srcToken, destToken);
+    if (!eventPool) return [];
+    return [eventPool.getIdentifer()];
+  }
+
+  async getPoolState(
+    pool: MakerPsmEventPool,
+    blockNumber: number,
+  ): Promise<PoolState> {
+    const eventState = pool.getState(blockNumber);
+    if (eventState) return eventState;
+    return await pool.generateState(blockNumber);
```

Comment:
If you generated a new state you should set it with `pool.setState`, then it can be used for future pricing requests on this block.